File: com.dexaint.dukechess/src/main/java/com/dexaint/dukechess/flow/Field.java
rt com.dexaint.dukechess.chess.Chess;
import com.dexaint.dukechess.chess.Chess;
import com.dexaint.dukechess.chess.ChessFactory;
import com.dexaint.dukechess.movement.MovementFactory;

public class Field {
	private int maxRow;
	public int getMaxRow(){return maxRow;}
	private int maxCol;
	public int getMaxCol(){return maxCol;}
	private Chess[] fieldMap;
	public Chess getChess(int pos) {return fieldMap[pos];}
	public void setChess(Chess chess, int pos) {fieldMap[pos]=chess;}
	
	private ChessFactory chessFactory = new ChessFactory();
	public ChessFactory getChessFactory() {return chessFactory;}
	private MovementFactory movementFactory = new MovementFactory();
	public MovementFactory getMovementFactory() {return movementFactory;}
	
	public Field(int maxRow, int maxCol)
	{
		this.maxRow=maxRow;
		this.maxCol=maxCol;
		this.fieldMap= new Chess[maxRow*maxCol];
		for(int i=0;i<maxRow*maxCol;i++)
				fieldMap[i]=null;
	}

}

File: jQuery-deprecated/public/js-backend/flow/Field.js
const ChessFactory = require("../chess/ChessFactory");
const MovementFactory = require("../movement/MovementFactory");

class Field {
	constructor(maxRow, maxCol)
	{
		this.maxRow=maxRow;
        this.maxCol=maxCol;
        
		this.fieldMap= [];
		for(var i = 0; i < this.maxRow * this.maxCol; i++)
            this.fieldMap[i] = null;
            
        this.chessFactory = new ChessFactory();
        this.movementFactory = new MovementFactory();
	}
}

module.exports = Field;